evm/db/account.py

```diff
@@ -65,6 +70,10 @@ def __init__(self) -> None:
     def state_root(self):
         raise NotImplementedError("Must be implemented by subclasses")
 
+    @abstractmethod
+    def has_root(self):
```

Comment:
This method signature doesn't match the implementation here: https://github.com/ethereum/py-evm/pull/704/files#diff-05a5232d6f5def53d9b58b9811d78282R218

Shouldn't this be: `def has_root(self, state_root: bytes) -> True:`?

> note I added type hints.